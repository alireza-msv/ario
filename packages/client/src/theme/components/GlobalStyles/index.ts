import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
  html {
    font-size: ${({ theme }) => theme.font.size};
    height: 100vh;
    height: -webkit-fill-available;
    height: -moz-available;
  }
  body {
    margin: 0;
    padding: 0;
    background-color: ${({ theme }) => theme.colors.background};
    font-family: ${({ theme }) => theme.font.family};
    font-weight: ${({ theme }) => theme.fontWeights.regular};
  }
  *, *:before, *:after {
    box-sizing: border-box;
  }
  a, a:visited {
    text-decoration: none;
    color: inherit;
  }
  h1, h2, h3, h4, h5, h6 {
    margin-top: 0;
    margin-bottom: 0;
    font-family: inherit;
  }
  #app {
    height: 100%;
  }
`;

export default GlobalStyles;
