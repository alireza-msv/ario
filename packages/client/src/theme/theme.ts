import { DefaultTheme } from 'styled-components';
import lightColors from './light';

const theme: DefaultTheme = {
  colors: lightColors,
  transitions: {
    fast: '150ms',
    normal: '300ms',
    slow: '500ms',
    superSlow: '1000ms',
  },
  font: {
    size: '16px',
    family: 'roboto',
  },
  fontWeights: {
    thin: 100,
    light: 300,
    regular: 400,
    medium: 500,
    bold: 700,
    black: 900,
  },
  zIndex: {
    zIndex0: 0,
    zIndex1: 1,
    zIndex10: 10,
    zIndex100: 100,
    zIndex1000: 1000,
  },
  breakpoints: {
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1600px',
  },
  borderRadiuses: {
    sm: '5px',
    md: '10px',
    lg: '15px',
    xl: '20px',
    xxl: '30px',
  },
  paddings: {
    none: '0',
    sm: '8px',
    md: '16px',
    lg: '24px',
    xl: '32px',
  },
  sizes: {
    layoutHorizontalPadding: '30px',
    layoutVerticalPadding: '30px',
    headerHeight: '104px',
  },
  grid: {
    cols: 12,
    gutter: '15px',
  },
};

export default theme;
