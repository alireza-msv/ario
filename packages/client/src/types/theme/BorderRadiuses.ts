interface BorderRadiuses {
  sm: string,
  md: string,
  lg: string,
  xl: string,
  xxl: string,
}

export default BorderRadiuses;
