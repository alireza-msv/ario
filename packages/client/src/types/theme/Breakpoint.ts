enum Breakpoint {
  SM = 'sm',
  MD = 'md',
  LG = 'lg',
  XL = 'xl',
  XXL = 'xxl',
}

export default Breakpoint;
