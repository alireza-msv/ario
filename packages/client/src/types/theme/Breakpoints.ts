interface Breakpoints {
  sm: string,
  md: string,
  lg: string,
  xl: string,
  xxl: string,
}

export default Breakpoints;
