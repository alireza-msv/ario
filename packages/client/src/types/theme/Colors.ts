interface Colors {
  primary500: string,
  secondary100: string,
  gray100: string,
  gray200: string,
  gray300: string,
  gray400: string,
  gray500: string,
  gray600: string,
}

export default Colors;
