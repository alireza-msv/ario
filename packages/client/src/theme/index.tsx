import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from './theme';
import Roboto from './fonts/Roboto';
import GlobalStyles from './components/GlobalStyles';

const Theme: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Roboto />
    <GlobalStyles />
    { children }
  </ThemeProvider>
);

export { theme };
export default Theme;
