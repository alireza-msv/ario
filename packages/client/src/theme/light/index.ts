import Colors from '~/types/theme/Colors';

const colors: Colors = {
  background: '#F7F7FA',
  foreground: '#FFF',
  primary500: '#BB2299',
  secondary100: '#F7F7FA',
  gray100: '#FAFAFA',
  gray200: '#BEBEBE',
  gray300: '#AAAAAA',
  gray400: '#757575',
  gray500: '#606060',
  gray600: '#404040',
};

export default colors;
