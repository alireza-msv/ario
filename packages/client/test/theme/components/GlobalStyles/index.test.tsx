import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import theme from '~/theme/theme.ts';
import GlobalStyles from '~/theme/components/GlobalStyles';
import 'jest-styled-components';

describe('GlobalStyles tests', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    render(
      <GlobalStyles theme={theme} />,
    );
    expect(document.head).toMatchSnapshot();
  });
});
