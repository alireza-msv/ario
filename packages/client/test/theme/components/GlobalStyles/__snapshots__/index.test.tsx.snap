// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyles tests should match snapshot 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.1.1"
  >
    html{font-size:16px;height:100vh;height:-webkit-fill-available;height:-moz-available;}
    body{margin:0;padding:0;background-color:#F7F7FA;font-family:roboto;font-weight:400;height:100%;}
    *,*:before,*:after{box-sizing:border-box;}
    a,a:visited{-webkit-text-decoration:none;text-decoration:none;color:inherit;}
    h1,h2,h3,h4,h5,h6{margin-top:0;margin-bottom:0;font-family:inherit;}
    #app{height:100%;}
    html{font-size:16px;height:100vh;height:-webkit-fill-available;height:-moz-available;}
    body{margin:0;padding:0;background-color:#F7F7FA;font-family:roboto;font-weight:400;height:100%;}
    *,*:before,*:after{box-sizing:border-box;}
    a,a:visited{-webkit-text-decoration:none;text-decoration:none;color:inherit;}
    h1,h2,h3,h4,h5,h6{margin-top:0;margin-bottom:0;font-family:inherit;}
    #app{height:100%;}
  </style>
</head>
`;
